### ISSUE TITLE
[Bug] Keys with ":" raise parsing error with parsing next blocks

### ISSUE THREAD

**Issue by Mr-S-Mirzoev** [biojppm/rapidyaml#379]
Good time of the day there. I believe I'm able to reproduce the bug:

```yaml
c: "proxy"
a:p_p:
  c: "clean"
  m: !<SomeClass> [/some/path]
a:d_d:
  c: "proxy"
  m: !<SomeClass> [/some/path]
```

<img width="1680" alt="image" src="https://github.com/biojppm/rapidyaml/assets/61053551/ade22435-8e99-4f16-b3a0-4502b3a786d6">

```log
-:8:4: (48B): ERROR: ': ' is not a valid token in plain flow (unquoted) scalars
-:8:4:   c: "clean"  (size=12)
          ^~~~~~~~~  (cols 4-13)
```

Indeed, it is some issue with keys with colons since the same yaml with keys without colons is parsed successfully by RYML:

```yaml
c: "proxy"
p_p:
  c: "clean"
  m: !<SomeClass> [/some/path]
d_d:
  c: "proxy"
  m: !<SomeClass> [/some/path]
```

<img width="1680" alt="image" src="https://github.com/biojppm/rapidyaml/assets/61053551/c5804e35-1b2b-4c74-8ccd-c4e018f431a7">

I also noticed that if you quote the keys, RYML parses successfully too.

```yaml
c: "proxy"
"a:p_p":
  c: "clean"
  m: !<SomeClass> [/some/path]
"a:d_d":
  c: "proxy"
  m: !<SomeClass> [/some/path]
```

<img width="1680" alt="image" src="https://github.com/biojppm/rapidyaml/assets/61053551/611bc8e8-f1d6-4c4a-84ef-d46db8e1e5ea">

Also, if this helps, the library acts weird and fails in this example only on the second element with a colon-containing key:

```yaml
a:p_p:
  c: "clean"
  m: !<SomeClass> [/some/path]
c: "text"
a:d_d:
  c: "proxy"
  m: !<SomeClass> [/some/path]
```

<img width="1680" alt="image" src="https://github.com/biojppm/rapidyaml/assets/61053551/e97d8d4d-7d90-4dbf-a3a3-c628faed2f07">

In case you don't see comments on this closed issue, let me mention you: @biojppm

_Originally posted by @Mr-S-Mirzoev in https://github.com/biojppm/rapidyaml/issues/377#issuecomment-1587943459_

**Comment by biojppm** [comment]
Thanks for the detailed description, it is helpful. Please hold for a fix (and expect some weeks for it to appear).

**Comment by biojppm** [comment]
This is likely fixed in the incoming parser refactor, #414.
